src/utilities/buildASTSchema.js

```diff
@@ -127,6 +130,43 @@ function getNamedTypeNode(typeNode: TypeNode): NamedTypeNode {
  * has no resolve methods, so execution will use default resolvers.
  */
 export function buildASTSchema(ast: DocumentNode): GraphQLSchema {
+  const config: any = buildASTSchemaComponents(ast);
```

Comment:
I believe you can fix the any check with:
```
const config = buildASTSchemaComponents(ast);
if (!config.query) {
  throw new GraphQLError(
    'Must provide schema definition with query type or a type named Query.',
    ast,
  );
}
...
return new GraphQLSchema(config);
```

I believe the issue for flow is that `config.query` can be `undefined` in addition to `null`.